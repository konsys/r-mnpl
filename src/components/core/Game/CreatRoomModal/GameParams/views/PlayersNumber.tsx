import { Button, ButtonGroup, Grid } from "@material-ui/core";

import { IRoomSetup } from "components/core/Game/FindGames/FindGame";
import React from "react";
import { useTranslation } from "react-i18next";

export default function PlayersNumber({
  setup,
  battleClosed,
}: {
  setup: IRoomSetup;
  battleClosed?: boolean;
}) {
  const { t } = useTranslation();
  return (
    <Grid
      container
      spacing={1}
      justify="space-between"
      alignItems="center"
      direction="row"
    >
      <Grid item>{t("Players")}</Grid>
      <Grid item className="playersNumber">
        <ButtonGroup disableElevation>
          <Button
            size="small"
            disabled={false}
            variant="contained"
            color={setup.state.playersNumber === 2 ? "primary" : "default"}
            onClick={() => setup.setState({ ...setup.state, playersNumber: 2 })}
          >
            2
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={setup.state.playersNumber === 3 ? "primary" : "default"}
            onClick={() => setup.setState({ ...setup.state, playersNumber: 3 })}
          >
            3
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={setup.state.playersNumber === 4 ? "primary" : "default"}
            onClick={() => setup.setState({ ...setup.state, playersNumber: 4 })}
          >
            4
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={setup.state.playersNumber === 5 ? "primary" : "default"}
            onClick={() => setup.setState({ ...setup.state, playersNumber: 5 })}
          >
            5
          </Button>
          {!battleClosed && (
            <Button
              disabled={false}
              variant="contained"
              color={setup.state.playersNumber === 6 ? "primary" : "default"}
              onClick={() =>
                setup.setState({ ...setup.state, playersNumber: 6 })
              }
            >
              2x2
            </Button>
          )}
        </ButtonGroup>
      </Grid>
    </Grid>
  );
}
