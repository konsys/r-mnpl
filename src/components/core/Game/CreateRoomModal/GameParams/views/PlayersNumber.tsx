import { Button, ButtonGroup, Grid } from "@material-ui/core";
import {
  preparatoryRoom$,
  setPreparatoryRoom,
} from "stores/Game/Rooms/RoomsModel";

import React from "react";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

export default function PlayersNumber({
  battleClosed,
  isVip,
}: {
  battleClosed?: boolean;
  isVip: boolean;
}) {
  const { t } = useTranslation();
  const room = useStore(preparatoryRoom$);
  return (
    <Grid
      container
      spacing={1}
      justify="space-between"
      alignItems="center"
      direction="row"
    >
      <Grid item>{t("Players")}</Grid>
      <Grid item className="playersNumber">
        <ButtonGroup disableElevation>
          <Button
            size="small"
            disabled={false}
            variant="contained"
            color={room.playersNumber === 2 ? "primary" : "default"}
            onClick={() => setPreparatoryRoom({ ...room, playersNumber: 2 })}
          >
            2
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={room.playersNumber === 3 ? "primary" : "default"}
            onClick={() => setPreparatoryRoom({ ...room, playersNumber: 3 })}
          >
            3
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={room.playersNumber === 4 ? "primary" : "default"}
            onClick={() => setPreparatoryRoom({ ...room, playersNumber: 4 })}
          >
            4
          </Button>
          <Button
            disabled={false}
            variant="contained"
            color={room.playersNumber === 5 ? "primary" : "default"}
            onClick={() => setPreparatoryRoom({ ...room, playersNumber: 5 })}
          >
            5
          </Button>
          {!battleClosed && (
            <Button
              disabled={!isVip}
              variant="contained"
              color={room.playersNumber === 6 ? "primary" : "default"}
              onClick={() => setPreparatoryRoom({ ...room, playersNumber: 6 })}
            >
              2x2
            </Button>
          )}
        </ButtonGroup>
      </Grid>
    </Grid>
  );
}
