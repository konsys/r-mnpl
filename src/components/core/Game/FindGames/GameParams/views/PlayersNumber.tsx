import { Button, ButtonGroup } from "@material-ui/core";

import { IRoomSetup } from "../../CreateGameModal";
import React from "react";

export default function PlayersNumber({ setup }: { setup: IRoomSetup }) {
  return (
    <ButtonGroup disableElevation>
      <Button
        size="small"
        disabled={false}
        variant="contained"
        color={setup.state.playersNumber === 2 ? "primary" : "default"}
        onClick={() => setup.setState({ ...setup.state, playersNumber: 2 })}
      >
        2
      </Button>
      <Button
        disabled={false}
        variant="contained"
        color={setup.state.playersNumber === 3 ? "primary" : "default"}
        onClick={() => setup.setState({ ...setup.state, playersNumber: 3 })}
      >
        3
      </Button>
      <Button
        disabled={false}
        variant="contained"
        color={setup.state.playersNumber === 4 ? "primary" : "default"}
        onClick={() => setup.setState({ ...setup.state, playersNumber: 4 })}
      >
        4
      </Button>
      <Button
        disabled={false}
        variant="contained"
        color={setup.state.playersNumber === 5 ? "primary" : "default"}
        onClick={() => setup.setState({ ...setup.state, playersNumber: 5 })}
      >
        5
      </Button>
      <Button
        disabled={false}
        variant="contained"
        color={setup.state.playersNumber === 6 ? "primary" : "default"}
        onClick={() => setup.setState({ ...setup.state, playersNumber: 6 })}
      >
        2x2
      </Button>
    </ButtonGroup>
  );
}
