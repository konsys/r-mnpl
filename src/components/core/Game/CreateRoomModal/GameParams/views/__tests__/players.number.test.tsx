import PlayersNumber from "../PlayersNumber";
import React from "react";
import { shallow } from "enzyme";

describe("Players number test", () => {
  it("should render vip", () => {
    expect(shallow(<PlayersNumber isVip />)).toMatchSnapshot();
  });
  it("should render not vip", () => {
    expect(shallow(<PlayersNumber isVip={false} />)).toMatchSnapshot();
  });
});
