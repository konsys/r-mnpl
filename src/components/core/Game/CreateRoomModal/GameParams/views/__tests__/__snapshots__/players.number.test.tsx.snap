// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players number test should render not vip 1`] = `ShallowWrapper {}`;

exports[`Players number test should render vip 1`] = `ShallowWrapper {}`;
